processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -549,10 +553,10 @@ private File makeIndexFiles(
     IOPeon ioPeon = new TmpFileIOPeon();
     ArrayList<FileOutputSupplier> dimOuts = Lists.newArrayListWithCapacity(mergedDimensions.size());
     Map<String, Integer> dimensionCardinalities = Maps.newHashMap();
-    ArrayList<Map<String, IntBuffer>> dimConversions = Lists.newArrayListWithCapacity(indexes.size());
+    ArrayList<Map<String, int[]>> dimConversions = Lists.newArrayListWithCapacity(indexes.size());
```

Comment:
Don't think it will affect performace (I'll check that tomorrow). Is IntBuffer is better than int[]? Less intuitive for me.